metadata/db.go

```diff
@@ -403,9 +409,11 @@ func (m *DB) GarbageCollect(ctx context.Context) (gc.Stats, error) {
 
 // getMarked returns all resources that are used.
 func (m *DB) getMarked(ctx context.Context, c *gcContext) (map[gc.Node]struct{}, error) {
+	ctx, span := tracing.StartSpan(ctx, tracing.Name(dbSpanPrefix, "getMarked"))
```

Comment:
Do we need a new span here? Based on multiple feedback, creation of new spans should mark entry/exit into a component,  for internal methods we can use the existing span in context (using `SpanFromContext()`) and add attributes/events accordingly. We can make exceptions based on the usecase, say while tracing methods that are known to have high latencies or go routines etc. In this scenario, I don't think creating a new span adds much value. Applies to other places as well. 

Also should keep in mind that some trace services charge by the number of spans, so adding a lot of spans might not be feasible